servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/SubscriberUtils.java

```diff
@@ -77,6 +77,19 @@ public static IllegalArgumentException newExceptionForInvalidRequestN(long n) {
         return new IllegalArgumentException("Rule 3.9 states non-positive request signals are illegal, but got: " + n);
     }
 
+    /**
+     * Create a new exception when a subscriber throws when it doesn't return "normally" according to
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm#2.13">Reactive Streams, Rule 2.13</a>.
+     * @param cause The original cause that was thrown.
+     * @return The exception which clarifies the invalid behavior.
+     */
+    public static RuntimeException newExceptionNonNormalReturn(Throwable cause) {
+        return SubscriberReturnNormalException.class.equals(cause.getClass()) ?
+                (SubscriberReturnNormalException) cause :
+                new IllegalStateException("Rule 2.13 states Subscriber methods must return normally. Operators that " +
```

Comment:
If this is visible to the user, should the message include clarification what normally means? As stated in the spec " The only legal way to signal failure to a Subscriber is via the onError method."

Reply:
I clarified the exception message.